pandas/io/parquet.py

```diff
@@ -181,7 +181,8 @@ def write(
             is_dir=partition_cols is not None,
         )
         if isinstance(path_or_handle, io.BufferedWriter):
```

Comment:
Could probably combine the two if conditions into one (if mypy accepts that)?

Reply:
I have combined the two conditions together now and changed the hasattr check to name attribute which resolved the issue when I run mypy pandas.